### DISCUSSION TITLE
Refactoring Activities (1.0.0)

### DISCUSSION THREAD

**Discussion by steven-bellock** [DMTF/libspdm#248] [Announcements]
Starting on 8-Nov-2021 and ending no later than 15-Nov-2021 we will begin a major refactor of libspdm in preparation for the first stable release. See the attached presentation for what will be changed.

**Please do not create any pull requests during this time.**

[DMTF_libspdm_refactor.pptx](https://github.com/DMTF/libspdm/files/7454635/DMTF_libspdm_refactor.pptx)

**Comment by steven-bellock** [comment]
Refactoring will begin this week. Please do not create any pull requests.

**Comment by jyao1** [comment]
9d8a8bb is ready.

**Comment by steven-bellock** [comment]
libspdm has most of its fixed-width integer types converted to the stdint.h types. Currently libspdm compiles on Linux (GCC) with both Mbed TLS and OpenSSL. On Windows only Mbed TLS can be compiled and the OpenSSL build is broken. This will be resolved in future pull requests.

**Comment by jyao1** [comment]
tag 1.0.0 (https://github.com/DMTF/libspdm/releases/tag/1.0.0) covers this.
Do we close this discussion?

**Reply by steven-bellock** [reply]
Doesn't look like we can close it. We can use a another thread for refactoring activities.
